java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -2107,14 +2113,17 @@ private ExpressionTree applySelectors(ExpressionTree primary, Optional<List<Expr
       for (ExpressionTree selector : selectors.get()) {
         if (selector.is(Kind.IDENTIFIER)) {
           IdentifierTreeImpl identifier = (IdentifierTreeImpl) selector;
-          result = new MemberSelectExpressionTreeImpl(result, identifier,
-            (AstNode) result, identifier);
+          // TODO The dot should have already been correctly handled
```

Comment:
At the moment I'm not able to find an way to get the `dot` more properly, as it has been introduced in the child of the `identifierTree`...